functional_requirements.md

```diff
@@ -0,0 +1,55 @@
+# Haveno Functional Requirements
+
+This document maintains functional requirements for the Haveno user interface (work in progress).
+
+### Startup
+- On first startup, the user must configure a password.
+- On first startup, the user can use a local node or configure a remote node (url, username, password).
```

Comment:
```suggestion
- On first startup, the user is asked if they want to use a local node or a remote one. If they choose 'local' Haveno will look for a local node. If they choose 'remote', they are prompted to another choice: 'Automatically detect' node or 'specify node' (defaults to Haveno public node) where they specify url + port, username (not mandatory), password (not mandatory)
```

Reply:
When selecting the option to use a local node, it sets up a node on the user's machine, right? 

In the case of using a remote node: how can we automatically detect which remote node the user wants to use? What is this based on?

Reply:
> When selecting the option to use a local node, it sets up a node on the user's machine, right?

No. It will look for a local node that is already running locally.

> how can we automatically detect which remote node the user wants to use? What is this based on?

The Monero GUI and Monerujo support scanning the network looking for valid remote nodes. We are considering implementing Monerujo's version of this feature (which is already in java) in the mvp, but it's not certain. What's certain is that there will be a hard coded node address that people will be able to change if they want to rely on a specific remote node.

Reply:
Got it.